File: Flowcharts/ConCalTwoIntNum/Cpp/ConCalTwoIntNum.cpp
#include <iostream>
#include <sstream>
#include <string>
#include <cstdlib>
#include <cmath>

using namespace std;

// Headers
string toString (double);
int toInt (string);
double toDouble (string);

int main() {
    int n, m, a;
    string cmd, more;
    bool goodtogo, doitagain;
    
    doitagain = true;
    while (doitagain == true) {
        cout << "Enter n value" << endl;
        cin >> n;
        cout << "Enter m value" << endl;
        cin >> m;
        goodtogo = false;
        while (goodtogo == false) {
            cout << "Enter command" << endl;
            cin >> cmd;
            if (cmd == "add") {
                a = n + m;
                goodtogo = true;
            }
            if (cmd == "sub") {
                a = n - m;
                goodtogo = true;
            }
            if (cmd == "mul") {
                a = m * n;
                goodtogo = true;
            }
            if (goodtogo == false) {
                cout << "Bad command" << endl;
            }
        }
        cout << "The answer is " << a << endl;
        cout << "Do you want to Calculate again? (yes or no)" << endl;
        cin >> more;
        if (more == "yes") {
            doitagain = true;
        } else {
            doitagain = false;
        }
    }
    cout << "The program is done" << endl;
    return 0;
}

// The following implements type conversion functions.
string toString (double value) { //int also
    stringstream temp;
    temp << value;
    return temp.str();
}

int toInt (string text) {
    return atoi(text.c_str());
}

double toDouble (string text) {
    return atof(text.c_str());
}

File: Flowcharts/ConCalTwoIntNum/Pascal/ConCalTwoIntNum.pas
program MyProgram;
uses Math, SysUtils;

var
    n, m, a : integer;
    Cmd, More : string;
    Goodtogo, Doitagain : boolean;

begin
    Doitagain := true;
    while Doitagain = true do
    begin
        WriteLn('Enter n value');
        ReadLn(n);
        WriteLn('Enter m value');
        ReadLn(m);
        Goodtogo := false;
        while Goodtogo = false do
        begin
            WriteLn('Enter command');
            ReadLn(Cmd);
            if Cmd = 'add' then
            begin
                a := n + m;
                Goodtogo := true
            end;
            if Cmd = 'sub' then
            begin
                a := n - m;
                Goodtogo := true
            end;
            if Cmd = 'mul' then
            begin
                a := m * n;
                Goodtogo := true
            end;
            if Goodtogo = false then
            begin
                WriteLn('Bad command')
            end
        end;
        WriteLn('The answer is ' + FloatToStr(a));
        WriteLn('Do you want to Calculate again? (yes or no)');
        ReadLn(More);
        if More = 'yes' then
        begin
            Doitagain := true
        end
        else
        begin
            Doitagain := false
        end
    end;
    WriteLn('The program is done')
end.
